src/components/atoms/Text/index.tsx

```diff
@@ -0,0 +1,244 @@
+          : undefined,
+      }),
+      [lineHeight, finalFontSize]
+    )
+
+    const letterSpacingStyle = useMemo<TextStyle>(
+      () => ({
+        letterSpacing: letterSpacing
+          ? convertEmToNumber(theme.letterSpacings[letterSpacing], finalFontSize)
+          : undefined,
+      }),
+      [letterSpacing, finalFontSize]
+    )
+
+    const textColor = useMemo<TextStyle>(
+      () => ({
+        color: color ? getColorValue({ color, colors: theme.colors }) : undefined,
+      }),
+      [color]
+    )
+
+    const textAlignmentStyle = useMemo<TextStyle>(
+      () => ({
+        textAlign: props.textAlign,
+      }),
+      [props.textAlign]
+    )
+
+    const textTransformStyle = useMemo<TextStyle>(
+      () => ({
+        textTransform:
+          capitalize || textTransform === 'capitalize'
+            ? 'capitalize'
+            : lowercase || textTransform === 'lowercase'
+            ? 'lowercase'
+            : uppercase || textTransform === 'uppercase'
+            ? 'uppercase'
+            : 'none',
+      }),
+      [capitalize, lowercase, uppercase, textTransform]
+    )
+
+    const textDecorationStyle = useMemo<TextStyle>(
+      () => ({
+        textDecorationLine:
+          underline || textDecoration === 'underline'
+            ? 'underline'
+            : strikeThrough || textDecoration === 'line-through'
+            ? 'line-through'
+            : undefined,
+      }),
+      [underline, strikeThrough, textDecoration]
+    )
+
+    const fontWeightStyle = useMemo<TextStyle>(
+      () => ({
+        fontWeight: (bold && 'bold') || (fontWeight && getFontWeight(fontWeight)),
+      }),
+      [fontWeight, bold]
+    )
+
+    const fontFamilyStyle = useMemo<TextStyle>(
+      () => ({
+        fontFamily: fontFamily ? theme.fonts[fontFamily] : undefined,
+      }),
+      [fontFamily]
+    )
+
+    const fontSizeStyle = useMemo<TextStyle>(
+      () => ({
+        fontSize: finalFontSize,
+      }),
+      [finalFontSize]
+    )
+
+    const textStyle = useMemo(
+      () =>
+        generateStyleSheet<TextStyle>([
+          bold && { fontWeight: 'bold' },
+          italic && { fontStyle: 'italic' },
+          fontFamilyStyle,
+          fontSizeStyle,
+          textAlignmentStyle,
+          textColor,
+          textDecorationStyle,
+          letterSpacingStyle,
+          lineHeightStyle,
+          textTransformStyle,
+          fontWeightStyle,
+          style,
+        ]),
+      [
+        bold,
+        italic,
+        fontFamilyStyle,
+        fontSizeStyle,
+        textAlignmentStyle,
+        letterSpacingStyle,
+        lineHeightStyle,
+        textColor,
+        textDecorationStyle,
+        textTransformStyle,
+        fontWeightStyle,
+        style,
+      ]
+    )
+
+    return <BaseText testID="baseText" numberOfLines={noOfLines} {...props} style={textStyle} />
+  }
+)
+
+export type TextVariant =
+  | 'H1'
+  | 'H1Regular'
```

Comment:
What is a difference between H1 and H1Regular, do we really need 3 types of single heading?

Maybe we can leave only two:
- H1Regular
- H1Bold

Reply:
Just to provide more flexibility to the user. Can remove

Reply:
It's up to you, IMO it's not needed but for now it can stay like that, so you don't need to spend time on that.